### PR TITLE
Replace class literals with reified methods

### PR DESCRIPTION
**Pull request by swankjesse**
To help, introduce our own assertThrows() method that works better with Kotlin

### CODE REVIEW

File: samples/urlshortener/src/test/kotlin/com/squareup/urlshortener/ShortenUrlWebActionTest.kt [hunk 1]
```diff
@@ -39,14 +39,14 @@ class ShortenUrlWebActionTest {
   @Test
   fun malformedLongUrl() {
     val longUrl = ""
-    assertThrows(BadRequestException::class.java) {
+    assertThrows<BadRequestException> {
```
Comment (ryanhall07) [inline samples/urlshortener/src/test/kotlin/com/squareup/urlshortener/ShortenUrlWebActionTest.kt]: nice

File: misk/testing/src/main/kotlin/misk/testing/JUnit.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package misk.testing
+
+/** Confirms that `block` throws `T`, and returns what was thrown. */
+inline fun <reified T : Throwable> assertThrows(block: () -> Unit): T {
```
Comment (ryanhall07) [inline misk/testing/src/main/kotlin/misk/testing/JUnit.kt]: where do we draw the line on the number of assertion helpers we add to misk? don't want it turning into something like assertj (which you could just use)

Comment (swankjesse) [reply misk/testing/src/main/kotlin/misk/testing/JUnit.kt]: We should be comfortable adding small APIs to make Kotlin nicer. All such APIs should be candidates for upstreaming to their respective projects.

Comment (swankjesse) [reply misk/testing/src/main/kotlin/misk/testing/JUnit.kt]: (I'd like to submit PRs to create a guice-kotlin module with our Guice.kt stuff, for example.)

**Comment by swankjesse** [comment]
Test failures appear to be a long-fixed compiler bug. Will upgrade to Kotlin 1.2.50 and rebase on that.
https://github.com/square/misk/pull/269

**Pull request merged**